src/components/Popup/Popup.tsx

```diff
@@ -126,16 +127,45 @@ export default class Popup extends AutoControlledComponent<Extendable<PopupProps
 
   private static isBrowserContext = isBrowser()
 
+  private clickSubscription = EventStackSubscription.empty()
+  private triggerDomElement = null
+
   protected actionHandlers: AccessibilityActionHandlers = {
-    toggle: e => this.trySetOpen(!this.state.open, e, true),
-    closeAndFocusTrigger: e => {
-      this.trySetOpen(false, e, true)
-      _.invoke(this.state.target, 'focus')
+    toggle: e => {
+      this.trySetOpen(!this.state.open, e, true)
     },
+    closeAndFocusTrigger: e => this.closeAndFocusTrigger(e),
+  }
+
+  private closeAndFocusTrigger = e => {
+    if (this.state.open) {
+      this.trySetOpen(false, e, true)
+      _.invoke(this.triggerDomElement, 'focus')
+    }
+  }
+
+  private closeAndFocusTriggerOnClickIfOpen() {
+    if (this.state.open) {
+      setTimeout(() => this.clickSubscription.replaceWith('click', this.closeAndFocusTrigger))
+    } else {
+      this.clickSubscription.stop()
+    }
   }
 
   public state = { target: undefined, open: false }
 
+  public componentDidMount() {
+    this.closeAndFocusTriggerOnClickIfOpen()
+  }
+
+  public componentDidUpdate() {
+    this.closeAndFocusTriggerOnClickIfOpen()
```

Comment:
what was a reason to have subscription logic on```componentDidUpdate```?  Isn't it enough to have it on the edge lifecycles callbacks, like ```componentDidMount``` and ```componentWillUnmount```?

Reply:
there are two subtle problems that might occur with the alternative approach (if we will subscribe only once).
- first is that we will have subscription for all the popups on the page, even if those are not opened. In other words, say, we will have 100 components on the page that are 'poppable', and this will result in 100 subscriptions _just because these components exist_. This introduces some perf problems, but, what is more important, changes to global app state that are hard to debug - when you are looking on the event listeners, you would expect at each moment of time that there are only those that necessary, and now, in contrast, you will have full fledge of subscriptions that exist for no good reason (if popup is closed - due to those are responsible for Popup behavior in open state)
- another one is maybe even more serious. If you'll try to implement these changes on your local machine, you will immediately see that popup is never opened on click. The problem is that _if client code forgets to call `e.preventDefault()`, event will 
  - set Popup state to `open`
  - then propagate to `document` element and execute the logic of handler, which says that `for open state close the Popup`
Thus, there won't be any changes visible to the client. Is it a problem of the client? I would say that certainly not, as _this event subscription should only occur when popup is opened_ - and this is what code provided is doing

Reply:
got your point, thank you very much for a detailed explanation